src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -1,11 +1,162 @@
+    private final WPI_TalonFX flywheelMotor2;
+    private final WPI_TalonFX turretMotor;
+    private final WPI_TalonFX hoodMotor;
+
+    // Constants
+    public static final double FLYWHEEL_VELOCITY = 10;
+    public static final double STOP_MOTOR = 0;
+    public static final double DEGREES_TO_ENCODER_TICKS = 2048 / 360; // 2048 ticks per 360 degrees
+    public static final double MIN_TURRET_ANGLE = -180; // Total ~360 degrees of rotation, assumes 0 is center
+    public static final double MAX_TURRET_ANGLE = 180;
+    public static final double TURRET_MAX_SPEED = 0.1;
+    public static final double TURRET_MIN_SPEED = -0.1;
+    public static final double TURRET_P = 0.01; // Placeholder PID constants
+    public static final double TURRET_I = 0;
+    public static final double TURRET_D = 0;
+    public static final double MAX_HOOD_ANGLE = 40.0;
+    public static final double MIN_HOOD_ANGLE = 5;
+    public static final SupplyCurrentLimitConfiguration flywheelCurrentLimit = new SupplyCurrentLimitConfiguration(true,
+            40, 40, 500);
+    public static final SupplyCurrentLimitConfiguration turretCurrentLimit = new SupplyCurrentLimitConfiguration(true,
+            10, 10, 500);
+    public static final SupplyCurrentLimitConfiguration hoodCurrentLimit = turretCurrentLimit;
+
+    /**
+     * Constructor for shooter subsystem.
+     * 
+     * @param flywheelMotor1
+     *            The first motor connected to the flywheel
+     * 
+     * @param flywheelMotor2
+     *            The second motor connected to the flywheel
+     * 
+     * @param turretMotor
+     *            The motor that controls the horizontal rotation of the
+     *            turret
+     * 
+     * @param hoodMotor
+     *            The motor that controls the angle of the hood above the
+     *            turret
+     * 
+     */
+    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor,
+            WPI_TalonFX hoodMotor) {
+        // Motor configs
+        flywheelMotor1.configSupplyCurrentLimit(flywheelCurrentLimit);
+        flywheelMotor1.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor2.configSupplyCurrentLimit(flywheelCurrentLimit);
+        flywheelMotor2.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor1.setInverted(false);
+        flywheelMotor2.follow(flywheelMotor1);
+        flywheelMotor2.setInverted(InvertType.OpposeMaster);
+
+        turretMotor.configForwardSoftLimitThreshold(MAX_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS);
+        turretMotor.configReverseSoftLimitThreshold(MIN_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS);
+        turretMotor.configForwardSoftLimitEnable(true);
+        turretMotor.configReverseSoftLimitEnable(true);
+        turretMotor.configSupplyCurrentLimit(turretCurrentLimit);
+        turretMotor.setNeutralMode(NeutralMode.Brake);
+        turretMotor.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0);
```

Comment:
Recommend using `TalonFXConfiguration` and `configAllSettings(..)` here (and for the other motors) so everything not explicitly set is reset to the default value. Unlike software, things set on the motors persist across power cycles since they're written into the motor flash memory.

Reply:
I'm having a bit of troubling finding out how to set the neutral mode through `TalonFXConfiguration`, in the meantime I'm planning on using `configFactoryDefault` at the beginning to reset to the defaults. (It seems more common to use `configFactorDefault` and other config methods, for example [CrossTheRoadElec Java Talon FX Phoenix Examples](https://github.com/CrossTheRoadElec/Phoenix-Examples-Languages/tree/master/Java%20Talon%20FX%20(Falcon%20500)).)

Reply:
The `TalonFXConfiguration` is most useful if you are configuring multiple motors with the same configuration properties (or mostly overlapping properties). Resetting to the default & configuring specific values works fine too.